"make install" fails if ldoc is not available

Originally reported here: https://github.com/awesomeWM/awesome/issues/1127#issuecomment-250840247

I can reproduce this. After `sudo chmod 0 /usr/bin/ldoc`, running `make && make install DESTDIR=/tmp/awesome-install` fails with
```
CMake Error at cmake_install.cmake:76 (file):
  file INSTALL cannot find
  "/tmp/awesome/.build-ZweiLED-x86_64-linux-gnu-6.1.1/awesomerc.lua".
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -106,6 +106,9 @@ set(AWESOME_C_FLAGS
 mark_as_advanced(AWESOME_C_FLAGS)
 target_compile_options(${PROJECT_AWE_NAME} PRIVATE ${AWESOME_C_FLAGS})
 
+# Make sure awesomerc.lua is generated
+add_dependencies(${PROJECT_AWE_NAME} generate_awesomerc)
+
 # Linux w/ GCC requires -rdynamic to get backtrace to fully work.
 #
 # For "historical reasons", CMake adds the option to the linker flags

--- a/awesomeConfig.cmake
+++ b/awesomeConfig.cmake
@@ -351,6 +351,10 @@ add_custom_command(
         ${BUILD_DIR}/awesomerc.lua
 )
 
+# Create a target for the auto-generated awesomerc.lua
+add_custom_target(generate_awesomerc DEPENDS ${BUILD_DIR}/awesomerc.lua)
+
+
 #}}}
 
 # {{{ Copy additional files

